README.md

```diff
@@ -50,3 +50,39 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+
+
+///
+
+The swap function performs an array flip operation. When the input is the number of cities n, the time complexity is O(n)
+
+The calculateroutelength function needs to traverse the entire path and calculate the distance, and the time complexity is O(n)
+
+Since the maximum execution stop is set to 1000 times, the number of loops is O(1000n) which simplifies to O(n)
+
+If the number of iterations is not limited, the 2opt algorithm will stop when all possible exchanges cannot further reduce the path length. 
+
+For n cities, the worst case number of iterations is O(n^2)
```

Comment:
Still don't know how you're getting $n^2$ here -- this is contrary to what you say above.